1.20/README.md

```diff
@@ -0,0 +1,207 @@
+
+In case of Nginx container image, the application source code is typically
+either static HTML pages or configuration files.
+
+To support the Source-to-Image framework, important scripts are included in the builder image:
+
+* The `/usr/libexec/s2i/run` script is set as the default command in the resulting container image (the new image with the application artifacts).
+
+* The `/usr/libexec/s2i/assemble` script inside the image is run to produce a new image with the application artifacts. The script takes sources of a given application (HTML pages), Nginx configuration files, and places them into appropriate directories inside the image. The structure of nginx-app can look like this:
+
+**`./nginx.conf`**--
+       The main nginx configuration file
+
+**`./nginx-cfg/*.conf`**
+       Should contain all nginx configuration we want to include into image
+
+**`./nginx-default-cfg/*.conf`**
+       Contains any nginx config snippets to include in the default server block
+
+**`./nginx-start/*.sh`**
+       Contains shell scripts that are sourced right before nginx is launched
+
+**`./nginx-perl/*.pm`**
+       Contains perl modules to be use by `perl_modules` and `perl_require` directives
+
+**`./`**
+       Should contain nginx application source code
+
+
+Build an application using a Dockerfile
+---------------------------------------
+Compared to the Source-to-Image strategy, using a Dockerfile is a more
+flexible way to build an Nginx container image with an application.
+Use a Dockerfile when Source-to-Image is not sufficiently flexible for you or
+when you build the image outside of the OpenShift environment.
+
+To use the Nginx image in a Dockerfile, follow these steps:
+
+#### 1. Pull a base builder image to build on
+
+podman pull ubi8/nginx-120
+
+#### 2. Pull an application code
+
+An example application available at https://github.com/sclorg/nginx-container.git is used here. To adjust the example application, clone the repository.
+
+```
+git clone https://github.com/sclorg/nginx-container.git nginx-container
+cd nginx-container/examples/1.20/
+```
+
+#### 3. Prepare an application inside a container
+
+This step usually consists of at least these parts:
+
+* putting the application source into the container
+* moving configuration files to the correct place (if available in the application source code)
+* setting the default command in the resulting image
+
+For all these three parts, you can either set up all manually and use the `nginx` command explicitly in the Dockerfile ([3.1.](#31-to-use-own-setup-create-a-dockerfile-with-this-content)), or you can use the Source-to-Image scripts inside the image ([3.2.](#32-to-use-the-source-to-image-scripts-and-build-an-image-using-a-dockerfile-create-a-dockerfile-with-this-content). For more information about these scripts, which enable you to set-up and run the nginx daemon, see the "Source-to-Image framework and scripts" section above.
```

Comment:
As reading this sentence "For more information..." I'm not able to tell what you mean by the part "which enable you to". Consider rephrasing it.

Reply:
Rephrased